### PR TITLE
Add Rendering Per Frame; Rework Render Pass Per Frame

### PR DESCRIPTION
**Pull request by steve-the-edwards**

### CODE REVIEW

File: workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RuntimeConfig.kt [hunk 1]
```diff
@@ -20,19 +20,44 @@ public annotation class WorkflowExperimentalRuntime
...
+   * This version of the runtime will process all actions queued within [actionWaitMs] AND call
+   * render() after processing them (i.e. cascade up and down the tree) before passing the
+   * updated rendering to the UI or until [frameTimeoutMs] has passed, at which point it will
+   * pass the new rendering to the UI (in other words it can't process forever if actions are
+   * queueing faster than [actionWaitMs]).
+   *
+   * i.e. this is one "Rendering" per 'frame'.
+   */
+  @WorkflowExperimentalRuntime
+  public data class RenderingPerFrame(
```
Comment (rjrjr) [inline workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RuntimeConfig.kt:47]: Sorry to bike shed, but I'm having a hard time distinguishing btw. the two names. Perhaps:

- `ExhaustActionsBeforeRenderPass`
- `DropExtraRenderings`

File: workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowRunner.kt [hunk 2]
```diff
@@ -72,58 +72,61 @@ internal class WorkflowRunner<PropsT, OutputT, RenderingT>(
...
-      // pass off something to the UI!
-      while (processingResult == null && frameTimeLeft > 0) {
+      // Then we can process any more actions queued up until the frame timeout when we should
+      // pass off something to the UI! If processingResult != null, that means we have an output
+      // for the external system, so we must go back and render then as well. If the result
+      // is ActionsExhausted, there are no actions ready to process.
+      while (
+        processingResult == null &&
+        processingResult !is ActionsExhausted &&
+        frameTimeLeft > 0
```
Comment (rjrjr) [inline workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowRunner.kt:111]: `processingResult == null && processingResult !is ActionsExhausted` is always true.

File: workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RuntimeConfig.kt (hunk 1 above)
Comment (steve-the-edwards) [reply workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RuntimeConfig.kt:47]: Oh no, I thought these names were so smart. Maybe we should have a quick huddle to decide so it minimizes the thrash.

File: workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt [hunk 3]
```diff
@@ -74,11 +74,11 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
     Channel<WorkflowAction<PropsT, StateT, OutputT>>(capacity = UNLIMITED)
   private var state: StateT
 
-  private val context = RealRenderContext(
-    renderer = subtreeManager,
-    sideEffectRunner = this,
-    eventActionsChannel = eventActionsChannel
-  )
+  // private var context = RealRenderContext(
```
[the remark this answers is not in the document]
Comment (steve-the-edwards) [reply workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt:77]: This was temp testing I forgot to change back.

**Comment by steve-the-edwards** [comment]
This work is obselete. Went with a slightly different implementation for Conflate Stale Renderings runtime.

**Pull request closed without merging**